samples/algorithms/repeat-until-success/SimpleRUS.qs

```diff
@@ -0,0 +1,121 @@
+    /// If this succeeds, the program will have effectively applied an 
+    /// (I + i√2X)/√3 gate on the target qubit.
+    /// If this fails, the program reruns the circuit up to <limit> times.
+    operation CreateQubitsAndApplySimpleGate(
+        inputValue : Bool,
+        inputBasis : Pauli,
+        limit : Int
+    )
+    : ( Bool, Result, Int ) {
+        using (register = Qubit[2]) {
+            let (success, numIter) = ApplySimpleGate(
+                inputBasis, inputValue, limit, register);
+            let result = Measure([inputBasis], [register[1]]);
+            return (success, result, numIter);
+        }
+    }
+
+    /// # Summary
+    /// Apply (I + i√2X)/√3 on qubits using repeat until success algorithm.
+    ///
+    /// # Input
+    /// ## inputBasis
+    /// Pauli basis in which to prepare input qubit
+    /// ## inputValue
+    /// Boolean value for input qubit (true maps to One, false maps to Zero)
+    /// ## limit
+    /// Integer limit to number of repeats of circuit
+    /// ## register
+    /// Qubit register including auxiliary and target qubits
+    ///
+    /// # Output
+    /// Tuple of (success, numIter) where success = false if the number of 
+    /// iterations (numIter) exceeds the input <limit>
+    operation ApplySimpleGate(
+        inputBasis : Pauli,
+        inputValue : Bool,
+        limit : Int,
+        register : Qubit[]
+    )
+    : (Bool, Int) {
+        // Initialize results to One by default.
+        mutable success = false;
+        mutable numIter = 0;
+        if (inputValue) {
+            // Prepare target qubit in |0⟩ or |1⟩ state, depending on input value
+            if (inputValue) {
+                X(register[1]);
+            }
+            PrepareQubit(inputBasis, register[1]);
+        }
+
+        repeat {
+            // Assert valid starting states for all qubits
+            AssertMeasurement([PauliZ], [register[0]], Zero,
+             "Auxiliary qubit is not in |0⟩ state.");
+            AssertQubitIsInState(register[1], inputBasis, inputValue);
+            
+            if (ApplyAndMeasureSimpleRUSCircuit(register) == Zero) { //|0⟩
+                set success = true;
+            }
+
+            set success = success or (numIter >= limit);
+            set numIter = numIter + 1;
+            Reset(register[0]);
+        }
+        until (success);
+        return (success, numIter);
+    }
+
+    /// # Summary
+    /// Apply RUS circuit and measure auxiliary qubit in Pauli Z basis 
+    ///
+    /// # Input
+    /// ## register
+    /// Qubit register including auxiliary and target qubits
+    operation ApplyAndMeasureSimpleRUSCircuit(
+        register : Qubit[]
+    )
+     : Result {
+        H(register[0]);
+        T(register[0]);
+        CNOT(register[0], register[1]);
+        H(register[0]);
+        CNOT(register[0], register[1]);
+        T(register[0]);
+        H(register[0]);
+
+        return Measure([PauliZ], [register[0]]);
```

Comment:
See comment above; I think this could reasonably be a call to `MResetZ` instead, but if you choose to keep the reset out of this operation then I would be in favor of replacing this with `M` for simplicity.